### PR TITLE
Update pako typings

### PR DESCRIPTION
**Pull request by calebegg**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodeca.github.io/pako/
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: pako/index.d.ts [hunk 1]
```diff
@@ -1,61 +1,84 @@
-// Type definitions for pako 0.2.8
+// Type definitions for pako 0.1.4
```
Comment (cappellin) [inline pako/index.d.ts]: Mmm... maybe the version is 1.0.4

**Review by dcappellin — CHANGES_REQUESTED** [review]
I think there's only a typo in the index.d.ts, but other changes are fine. Good job!

File: pako/index.d.ts (hunk 1 above)
Comment (calebegg) [inline pako/index.d.ts]: Ah, good catch. Fixed.

File: pako/index.d.ts [hunk 2]
```diff
@@ -1,61 +1,84 @@
...
+		level?: number;
+		windowBits?: number;
+		memLevel?: number;
+		strategy?: number;
+		dictionary?: any;
+		raw?: boolean;
+		to?: 'string';
+	}
+
+  export interface InflateOptions {
```
Comment (zhengbli) [inline pako/index.d.ts]: indentation is off

File: pako/index.d.ts [hunk 3]
```diff
@@ -1,61 +1,84 @@
...
+	}
+
+	export type Data = Uint8Array | Array<number> | string;
+
 	/**
 	 * Compress data with deflate algorithm and options.
 	 */
-	export function deflate( data: Uint8Array | Array<number> | string, options?: any ): string;
+	export function deflate( data: Data, options: DeflateOptions & {to: 'string'} ): string;
+	export function deflate( data: Data, options?: DeflateOptions ): Uint8Array;
```
Comment (zhengbli) [inline pako/index.d.ts]: I guess you are trying to enforce `to` to be a required property if the `options` is provided, however the optional parameter here ignores that requirement, and TypeScript do not differentiate signatures just via return types. Removing the `options?` here might work

File: pako/index.d.ts [hunk 4]
```diff
@@ -1,61 +1,84 @@
...
+
 	/**
 	 * The same as deflate, but creates raw data, without wrapper (header and adler32 crc).
 	 */
-	export function deflateRaw( data: Uint8Array | Array<number> | string, options?: any ): string;
+	export function deflateRaw( data: Data, options: DeflateOptions & {to: 'string'} ): string;
+	export function deflateRaw( data: Data, options?: DeflateOptions ): Uint8Array;
```
Comment (zhengbli) [inline pako/index.d.ts]: same here

**Review by zhengbli — CHANGES_REQUESTED** [review]

File: pako/index.d.ts (hunk 3 above)
Comment (calebegg) [inline pako/index.d.ts]: No, I'm just trying to enforce that if `{to: 'string'}` is present, the return type is 'string'. [Playground link](http://www.typescriptlang.org/play/index.html#src=declare%20namespace%20Foo%20%7B%0D%0A%20%20%20%20interface%20Data%20%7B%20%7D%0D%0A%20%20%20%20interface%20DeflateOptions%20%7B%20%7D%0D%0A%20%20%20%20export%20function%20deflate(data%3A%20Data%2C%20options%3A%20DeflateOptions%20%26%20%7B%20to%3A%20'string'%20%7D)%3A%20string%3B%0D%0A%20%20%20%20export%20function%20deflate(data%3A%20Data%2C%20options%3F%3A%20DeflateOptions)%3A%20Uint8Array%3B%0D%0A%7D%0D%0A%0D%0Alet%20a%3A%20Uint8Array%20%3D%20Foo.deflate('a')%3B%0D%0Alet%20a2%3A%20Uint8Array%20%3D%20Foo.deflate('a'%2C%20%7B%20stuff%3A%20true%20%7D)%3B%0D%0Alet%20s%3A%20string%20%3D%20Foo.deflate('a'%2C%20%7B%20to%3A%20'string'%20%7D)%3B)

File: pako/index.d.ts (hunk 2 above)
Comment (calebegg) [inline pako/index.d.ts]: Oops. Not used to working with tabs :-)

Fixed.

**Review by zhengbli — APPROVED** [review]

**Comment by zhengbli** [comment]
@calebegg thanks!

**Pull request merged**